### DISCUSSION TITLE
How to view updates for an app?

### DISCUSSION THREAD

**Discussion by vodkadrunkinski** [ValvePython/steamctl#29] [Q&A] [answered]
Is it possible to use steamctl to view the updates for an app, using only the appid (or even a depot id)?

**Comment by vodkadrunkinski** [comment]
Possible answer (for Rust game)?
`steamctl webapi call ISteamNews.GetNewsForApp appid=252490 count=1 feeds=steam_community_announcements`

Not sure if that's the best solution to track game updates though...

**Comment by rossengeorgiev** [comment, marked as answer]
I'm assuming you are looking to check if there is an update for an app. Similar to this: https://github.com/ValvePython/steamctl/issues/26

Each app has a public branch, that's the default one that Steam downloads. Each branch can be one or more depots. Anyway, each branch has a `buildid`, so checking that can tell if there has been an update. Below is an example how to get the `buildid`:

```json
# steamctl --anonymous apps product_info 740 | jq -r .depots.branches.public.buildid
6193619
# steamctl --anonymous apps product_info 740 | jq .depots.branches
{
  "1.37.7.9": {
    "buildid": "6154962",
    "description": "1.37.7.9",
    "pwdrequired": "1",
    "timeupdated": "1612893933"
  },
  "1.37.8.0": {
    "buildid": "6185846",
    "description": "1.37.8.0",
    "timeupdated": "1612894215"
  },
....
```

You could write script to check the `buildid`, and determine if there is an update.

**Reply by vodkadrunkinski** [reply]
Thanks for the tip!

Here's the easy way to do the check (in bash):

```
UT=`steamctl --anonymous apps product_info 252490 | jq -r .depots.branches.public.timeupdated`
date -d @$UT
```
